pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/AbstractMetadataStore.java

```diff
@@ -246,6 +261,18 @@ public void registerSessionListener(Consumer<SessionEvent> listener) {
     }
 
     protected void receivedSessionEvent(SessionEvent event) {
+        switch (event) {
```

Comment:
this code is the same we added recently.
what about adding an utility method in SessionEvent 

```
public boolean isConnected() {
      switch (event) {
        case Reconnected:
        case SessionReestablished:
             isConnected = true;
             break;
        case ConnectionLost:
        case SessionLost:
             isConnected = false;
             break;
    }
    return isConnected;
}
```

Reply:
Yes, added